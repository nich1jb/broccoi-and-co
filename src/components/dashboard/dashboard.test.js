import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Dashboard from '.';

import 'mutationobserver-shim'
global.MutationObserver = window.MutationObserver

afterEach(cleanup);

describe('Dashboard', () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Dashboard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a title', () => {
    const { getByRole } = render(<Dashboard />)
    const title = getByRole('heading')
    expect(title.textContent).toBe('A better way to enjoy every day.')
  })

  it('clicking on the button will launch popup modal', () => {
    const { getByRole, getByText } = render(<Dashboard />)
    const button = getByRole('button')
    fireEvent.click(button)
    const modalTitle = getByText('Request an Invite')
    expect(modalTitle).toBeTruthy()
  })

  it('clicking outside modal will close it', () => {
    const { getByRole, queryByText } = render(<Dashboard />)
    const button = getByRole('button')
    fireEvent.click(button)
    const body = getByRole('generic', {name: 'modal-backdrop'})
    fireEvent.click(body)
    expect(queryByText('Request an Invite')).toBeNull()
  })
})