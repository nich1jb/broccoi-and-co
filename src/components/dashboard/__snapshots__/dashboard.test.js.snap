// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jrsJCI hRBrla"
  >
    <div
      class="sc-bdnylx kOKJeA"
    >
      <img
        class="sc-dlnjPT ksPreD"
        src="images/logo.png"
      />
    </div>
    <div
      class="sc-kEqYlL eiAWbY"
    >
      <h1
        class="sc-crzoUp gAWKtw"
      >
        A better way to enjoy every day.
      </h1>
      <div
        class="sc-dIsAE lbEpKq"
      >
        Be the first to know when we launch
      </div>
      <button
        class="sc-bqGHjH iNJljF"
      >
        Request an invite
      </button>
    </div>
    <div
      class="sc-iqAbSa jkfJHj"
    >
      <div
        class="sc-ksluoS itTrBN"
      >
        Photo by Lukas from Pexels
      </div>
      <div
        class="sc-bdnylx cLAbTx"
      >
        <div
          class="sc-gtssRu ieArEE"
        >
          Made with 
          <span
            aria-label="heart"
            role="img"
          >
            ðŸ’š
          </span>
           in Melbourne
        </div>
        <div
          class="sc-gtssRu ieArEE"
        >
          Â© 2016 Broccoli & Co. All Rights Reserved
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
